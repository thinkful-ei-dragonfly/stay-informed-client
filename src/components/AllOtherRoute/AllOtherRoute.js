import React from 'react';
import { Route } from 'react-router-dom';
import UserContext from '../../contexts/UserContext';

export default function AllOtherRoute({ component, ...props }) {
  const Component = component;
  return (
    <Route
      {...props}
      render={componentProps => (
        <UserContext.Consumer>
          {userContext => 
          <Component {...componentProps} />}
        </UserContext.Consumer>
        
      )}
    />
  );
}
