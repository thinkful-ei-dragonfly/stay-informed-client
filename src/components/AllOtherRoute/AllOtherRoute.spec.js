import React from 'react';
import { shallow } from 'enzyme'
import toJson from 'enzyme-to-json'
import AllOtherRoute from './AllOtherRoute'

describe(`AllOtherRoute component`, () => {
  it('renders a form.AllOtherRoute by default', () => {
    const wrapper = shallow(<AllOtherRoute />)
    expect(toJson(wrapper)).toMatchSnapshot()
  })
})
