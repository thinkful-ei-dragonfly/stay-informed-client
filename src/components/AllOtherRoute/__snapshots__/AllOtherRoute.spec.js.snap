// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AllOtherRoute component renders a form.AllOtherRoute by default 1`] = `
<Route
  render={[Function]}
/>
`;
